### DISCUSSION TITLE
Mimicking DRT with MVR in MF6

### DISCUSSION THREAD

**Discussion by javgs-bd** [MODFLOW-ORG/modflow6#1772] [Q&A] [answered]
I vaguely recall reading somewhere that the DRN return package functionality could be implemented through the MVR package.

I'm trying to simulate a situation where a trench has a maximum pumping flow capacity. This means, the trench should work as a DRN bc until a certain flow is reached, when it will not be able to extract it.

I have tried to replace the DRN bc with the EVT package, but this means defining a Qmax for each cell of the BC. It ends up limiting the outflow in simulations that where not supposed to reach the maximum flow. Changing DRN conductance would not work either, because this is an ensemble simulation, where aquifer K is stochastically changing close to the trench. 

I guess, one way to solve this is to keep the DRN bc but to reinject (in cells closeby to the trench) any bc flow over Qmax. For instance, in the lowest elevation cell of the BC. I imagine a  sort of feedback loop emerging where water heads increase due to reinjection and the DRN bc outflow equilibrates to Qmax.

Now, from the MVR docs I have this:
_The MVR Package can be used to transfer water from individual “provider” features of selected packages (WEL, DRN, RIV, GHB, MAW, SFR, LAK, and UZF) to individual “receiver” features of the advanced packages (MAW, SFR, LAK, and UZF). Simple rules are used to determine how much of the available water is moved from the provider to the receiver, which allows management controls to be represented._

Does this means I can only mimick DRT by using the DRN package, define a MVR and then reinject with MAW using an EXCESS rule?
I'm not seeing this case usage as an example in the 2021 Morway, Langevin, Hughes paper (still to read), and wonder if the MAW will add up any additional numerical burden for not being a WEL bc.

**Comment by emorway-usgs** [comment]
Hello @javgs-bd, I am unable to decipher what you are trying to do with the DRN package.  If, in a previous version of MODFLOW, the DRT package was able to represent your situation, then a combination of the DRN package with MVR connections should be able to simulate a similar setup.  

From what I can tell, you want to simulate an open trench with the DRN package, and when the flow entering the open trench exceeds a certain rate, you want to reinject the flow above a user-defined threshold back into the model domain with the MVR?  My confusion stems from what you plan to do with the water below the threshold?  MODFLOW will remove this water from the model and report it in the budget that is printed in the listing file.  Is that what you want?

One other matter for your consideration:  Is the open trench represented by more than one DRN boundary (cell)?  If what you intend is for the _combined flow_ from all drains representing the open trench to be subject to the threshold, that won't work as intended, I don't think.  Each individual DRN has to be linked with a 'receiver' using the MVR.  In other words, MVR works with as many individual connections as you would like, but a grouping of 'providers' cannot be evaluated relative to one of the four "rules" (`mvrtype`).  This is likely not what you are attempting to do, but I just wanted to make that nuance clear.

Again, I may not be understanding your specific use case, but it almost sounds to me like you want to simulate an open trench and once the stage within the trench exceeds a threshold it should somehow "bleed off" the excess?

**Reply by javgs-bd** [reply]
Thank.you very much for the quick reply @emorway-usgs .
Apologies if I was not clear enough.

Yes, I want to reinject the flow above a user-defined (Q) threshold back into the model domain.
The water below the threshold should be removed and printed as an outflow.

The trench is represented by a DRN package, that has a group of cells with DRN boundary conditions.  Let's say this trench is only 10 cells in one model row, in the middle of the domain, far from the model boundaries. The trench has a pump in it that can only evacuate Qmax. So, more than bleed off at a certain stage, it does so at a certain Q.

Therefore, I want the threshold to be applied to the combined (aggregated) flow of all the trench cells. I understood that that could be done with an EXCESS rule, as the 'providers' for the MVR are packages and not individual cells. 

I just got.a bit confused by your note on multiple providers. So, if the trench is represented by one single DRN package, am I right to understand it could be used as a single MVR provider, wich 'provides' the aggregated flow off all cells in the trench DRN package?

**Reply by javgs-bd** [reply]
I'm sorry, that last question should read "...that 'provides' the excess over Qmax of the aggregated flow of..."

**Comment by emorway-usgs** [comment, marked as answer]
Hello @javgs-bd, based on your description, I do think that using the MVR with the `EXCESS` rule will help meet your needs; however, the MVR cannot evaluate an aggregated flow from multiple providers.  I'm not sure if the image below will help, but it attempts to show what MVR can do on the left and what it cannot do on the right (which is what I gather you would like to do).  

In the example on the left, which shows a total of 5 adjacent DRN boundaries, MVR will support 1-to-1 connections (shown with "MVR1" and "MVR2") between a provider and a receiver (MAW boundaries, in this small example) or many-to-1 connections (shown with "MVR3", "MVR4", and "MVR5").  Labeling each connecting arrow with "MVR" attempts to highlight that each connection requires its own line in the MVR package (filled in some of the fields to correspond to the labels in the image below):

`BEGIN PERIOD <iper>`
`  [<mname1>] <pname1> DRN1 [<mname2>] <pname2> MAW1 EXCESS <value>`
`  [<mname1>] <pname1> DRN2 [<mname2>] <pname2> MAW2 EXCESS <value>`
`  [<mname1>] <pname1> DRN3 [<mname2>] <pname2> MAW3 EXCESS <value>`
`  [<mname1>] <pname1> DRN4 [<mname2>] <pname2> MAW3 EXCESS <value>`
`  [<mname1>] <pname1> DRN5 [<mname2>] <pname2> MAW3 EXCESS <value>`

The example on the right, which is my understanding of what you would like to do, depicts "DRN3", "DRN4", and "DRN5" being aggregated prior to being transferred to a receiver ("MAW3") by "MVR3".  MVR doesn't support aggregation in this way.  To achieve the setup you would like, you might need to consider use of the [API](https://doi.org/10.1016/j.envsoft.2021.105257).

![MVRs](https://github.com/MODFLOW-USGS/modflow6/assets/3236576/07fec300-c6d2-493d-9145-53f64b2b632e)

**Reply by javgs-bd** [reply]
Hi @emorway-usgs. I truly appreciate that detailed explanation.  
I see now what I was missing (even after reading the MVR paper). I think it was this line on the MF6 documentation: _The MVR Package works at the package feature level._ The key thing here being what a feature is within a package. I was reasoning as if providers and receivers could be _pakages_ and not _instances/features_ within packages. 

In my case a feature/instance of the drain package is a single cell, and not the entire package. 

After internal discussion here, we believe one solution to our problem (to duck the API need) is to implement a very simple child ancillary model that does the required WB and communicates with the parent through MVRs. This should readily work for our model which has only  steady state SPs.

This child model would have a MAW, a WEL, and a GHB boundary. The MAW in the child model receives (through a MVR connection between models) and injects the aggregate flow that results from all the DRN instances (cells) that represent the trech (i.e, one DRN package) in the parent model. The child WEL extracts the original Qmax constant (originally thought as EXCESS value). Thus the GHB boundary would only evacuate flow from the child aquifer when the MAW injection rate is above Qmax, i.e., the WEL extraction rate. That GHB outflow would be transferred back to the parent model with a second MVR and reinjected using a MAW. This uses the fact that only GHB outflow is available water for MVRs.

Sure the API could save some trouble. 
Cheers,
Javier

BTW: I think you meant MAWs in your diagram instead of WELs (which cannot be receivers)

**Reply by emorway-usgs** [reply]
> BTW: I think you meant MAWs in your diagram instead of WELs (which cannot be receivers)

Thanks for the correction @javgs-bd.  WEL cannot be a receiver.  I will update the figure in the post in case others happen across this post.

While I don't fully understand your child model setup, another option might be to pass the EXCESS flow from each DRN to an SFR reach, for example, and then setup a 1-to-1 connection between the SFR reach and MAW.  I'm not sure how well this would work, but if it did work the SFR "receiver" would serve as an 'aggregator' that then becomes the provider to MAW.  You could place the SFR reach anywhere in the model and would set the streambed hydraulic conductivity equal to 0.0 to ensure the reach doesn't interact with the aquifer.  The SFR-to-MAW MVR connection would use the FACTOR rule with the corresponding `value` being set equal to 1.0.

**Reply by javgs-bd** [reply]
Thanks!. 
I cannot apply the excess rule to each DRN cell, because they do not extract GW discharge uniformly. 
I could pass FACTOR 1 of each DRN cell to the SFR reach, and then, if it is possible, I could apply the excess rule on the SFR to pass only what exceeds Qmax (the capacity of the evacuation pump in the trench) to the MAW.

It seems a better option, since we are seeing that child model instabilities could get in the way.

**Comment by gschampion** [comment]
Have you considered using the multi-aquifer well (MAW) Package to represent the multi-cell trench using a SPECIFIED conductance? If you set the Q to the maximum Q and set a HEAD_LIMIT, I think it would behave like a drain with your specified conductance, but the Q would be limited to Qmax. If Qmax can't be achieved because of the head limit, Q will be less than Qmax.
